website/build_plugins.py

```diff
@@ -0,0 +1,167 @@
+                            print('Cannot evaluate value in '
+                                  + filepath + ':' +
+                                  ast.dump(node))
+        return data
+
+
+def build_json(source_dir, dest_dir):
+    """Traverse the plugins directory to generate json data."""
+
+    plugins = {}
+
+    # All top level directories in source_dir are plugins
+    for dirname in next(os.walk(source_dir))[1]:
+
+        files = {}
+        data = {}
+
+        if dirname in [".git"]:
+            continue
+
+        dirpath = os.path.join(source_dir, dirname)
+        for root, dirs, filenames in os.walk(dirpath):
+            for filename in filenames:
+                ext = os.path.splitext(filename)[1]
+
+                if ext not in [".pyc"]:
+                    file_path = os.path.join(root, filename)
+                    with open(file_path, "rb") as md5file:
+                        md5Hash = md5(md5file.read()).hexdigest()
+                    files[file_path.split(os.path.join(dirpath, ''))[1]] = md5Hash
+
+                    if ext in ['.py'] and not data:
+                        data = get_plugin_data(os.path.join(source_dir, dirname, filename))
+
+        if files and data:
+            print("Added: " + dirname)
+            data['files'] = files
+            plugins[dirname] = data
+    out_path = os.path.join(dest_dir, PLUGIN_FILE_NAME)
+    with open(out_path, "w") as out_file:
+        json.dump({"plugins": plugins}, out_file, sort_keys=True, indent=2)
+
+
+def zip_files(source_dir, dest_dir):
+    """Zip up plugin folders"""
+
+    for dirname in next(os.walk(source_dir))[1]:
+        archive_path = os.path.join(dest_dir, dirname)
+        archive = zipfile.ZipFile(archive_path + ".zip", "w")
+
+        dirpath = os.path.join(source_dir, dirname)
+        plugin_files = []
+
+        for root, dirs, filenames in os.walk(dirpath):
+            for filename in filenames:
+                file_path = os.path.join(root, filename)
+                plugin_files.append(file_path)
+
+        if len(plugin_files) == 1:
+            # There's only one file, put it directly into the zipfile
+            archive.write(plugin_files[0],
+                          os.path.basename(plugin_files[0]),
+                          compress_type=zipfile.ZIP_DEFLATED)
+        else:
+            for filename in plugin_files:
+                # Preserve the folder structure relative to source_dir
+                # in the zip file
+                name_in_zip = os.path.join(os.path.relpath(filename,
+                                                           source_dir))
+                archive.write(filename,
+                              name_in_zip,
+                              compress_type=zipfile.ZIP_DEFLATED)
+
+        print("Created: " + dirname + ".zip")
+
+
+def download_plugins(version=None):
+    """Downloads and extracts the plugin source files"""
+    temp_dir = mkdtemp(suffix="PICARD-WEBSITE")
```

Comment:
To be safer, i would enclose the following code in `try:` / `finally: shutil.rmtree(temp_dir)` block, as well in the places where `download_plugins()` is used.